ui/src/pages/users.vue

```diff
@@ -262,6 +364,56 @@ const autofill = ref({
   cas_id: "",
 });
 
+// New variables for delete confirmation modals
+const isDeleteModalVisible = ref(false);
+const isUsernamePromptVisible = ref(false);
+const usernameConfirmation = ref("");
+const isHardDelete = ref(false);
+const isSoftDeleted = computed(() => editedUser.value.is_deleted); //check soft delete status
+
+// Open the initial delete confirmation modal
+function confirmDeleteUser() {
+  isDeleteModalVisible.value = true;
+}
+
+const getImpactDetails = computed(() => {
```

Comment:
It's better to avoid naming computed properties like getters. So, let's rename this to `impactDetails` if possible, or some other name that does not contain the word `get`.